extmod/modtls_mbedtls.c

```diff
@@ -250,13 +252,30 @@ STATIC mp_obj_t ssl_context_make_new(const mp_obj_type_t *type_in, size_t n_args
 
     return MP_OBJ_FROM_PTR(self);
 }
+STATIC int ssl_sock_cert_verify(void *ptr, mbedtls_x509_crt *crt, int depth, uint32_t *flags) {
+    mp_obj_ssl_socket_t *o = ptr;
+    if (o->ssl_context->handler == mp_const_none) {
+        return 0;
+    }
+    return mp_obj_get_int(mp_call_function_2(o->ssl_context->handler, mp_obj_new_bytes(crt->raw.p, crt->raw.len), MP_OBJ_NEW_SMALL_INT(depth)));
```

Comment:
I suggest using `mp_obj_new_bytearray_by_ref(ctr->raw.len, crt->raw.p)` (instead of `mp_obj_new_bytes`).  Using a bytearray by reference will save RAM by not creating a copy of the data.

The downside of using bytearray by reference is that the object (memory it points to) becomes invalid after the callback.  But, this is an advanced feature and we can document the behaviour.  We need to save memory wherever possible.

Also, is it worth passing through the `*flags` argument??  I'm not sure what this does though.

Reply:
The flags argument is an in-out parameter that allows the inspection and modification of the "default" verification errors: https://github.com/Mbed-TLS/mbedtls/blob/32c28cebb41a25bb4e18d3a530b59c0e5929a324/include/mbedtls/x509.h#L91

Reply:
So I thought its more hassle and confusion to pass that through, than it would actually be worth.